crates/c-api/src/component/component.rs

```diff
@@ -85,3 +85,44 @@ pub unsafe extern "C" fn wasmtime_component_clone(
 
 #[unsafe(no_mangle)]
 pub unsafe extern "C" fn wasmtime_component_delete(_component: Box<wasmtime_component_t>) {}
+
+#[repr(transparent)]
+pub struct wasmtime_component_export_index_t {
+    pub(crate) export_index: ComponentExportIndex,
+}
+
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn wasmtime_component_get_export_index(
+    component: &wasmtime_component_t,
+    instance_export_index: *const wasmtime_component_export_index_t,
+    name: &mut wasm_name_t,
+    out: &mut *mut wasmtime_component_export_index_t,
+) -> bool {
+    let name = name.take();
+    let Ok(name) = String::from_utf8(name) else {
+        return false;
+    };
```

Comment:
Oh for this I think we'll want to have this be `const wasm_name_t*` in C and not take ownership here.

Also I apologize again for yet more whiplash... Looking at `wasmtime_linker_t` it looks like the functions there idiomatically take a pointer/length instead of `&wasm_name_t` (e.g. [`wasmtime_linker_define`](https://github.com/bytecodealliance/wasmtime/blob/5614b80b58d946a9a5270b1c56934fec99f097d9/crates/c-api/src/linker.rs#L51-L66)). I had forgotten about that where `wasm_name_t` I think is mostly used for return values as opposed to arguments. In light of that could the idioms of the component linker be similar where a pointer/length is given? (sorry I should have checked this earlier)